slashing-protection/src/main/java/tech/pegasys/web3signer/slashingprotection/DbSlashingProtection.java

```diff
@@ -86,8 +86,12 @@ public DbSlashingProtection(
             signedAttestationsDao,
             metadataDao,
             lowWatermarkDao);
+
     this.dbPruner =
-        new DbPruner(pruningJdbi, signedBlocksDao, signedAttestationsDao, lowWatermarkDao);
+        pruningJdbi != null
+            ? Optional.of(
+                new DbPruner(pruningJdbi, signedBlocksDao, signedAttestationsDao, lowWatermarkDao))
+            : Optional.empty();
```

Comment:
pruningJdbi is an optional but we're checking for nullability so don't think this will actually work?

Reply:
I don't like using two optionals for this, think it would be better with 1 optional by passing the optionally constructed DbPruner into this class, i.e. pull up the construction of the DbPruner class that calls this constructor.